hdfs-scheduler/src/test/java/org/apache/mesos/hdfs/TestHdfsState.java

```diff
@@ -0,0 +1,325 @@
+package org.apache.mesos.hdfs;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.util.concurrent.ExecutionException;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.apache.mesos.hdfs.scheduler.Task;
+import org.apache.mesos.hdfs.state.HdfsState;
+import org.apache.mesos.hdfs.util.HDFSConstants;
+import org.apache.mesos.hdfs.util.StatusFactory;
+import org.apache.mesos.Protos.CommandInfo;
+import org.apache.mesos.Protos.ExecutorID;
+import org.apache.mesos.Protos.ExecutorInfo;
+import org.apache.mesos.Protos.FrameworkID;
+import org.apache.mesos.Protos.Label;
+import org.apache.mesos.Protos.Labels;
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.OfferID;
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.SlaveID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskState;
+import org.apache.mesos.Protos.TaskStatus;
+import org.apache.mesos.Protos.Value;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class TestHdfsState {
+  private final Injector injector = Guice.createInjector(new TestSchedulerModule());
+  private final Log log = LogFactory.getLog(TestHdfsState.class);
+  private SecureRandom random = new SecureRandom();
+  private static final String testIdName = "framework";
+  private static final String TEST_HOST = "host";
+  private static final String TEST_TYPE = "type";
+  private static final String TEST_NAME = "name";
+
+  @Test
+  public void testSetFrameworkId()
+    throws ClassNotFoundException, ExecutionException, InterruptedException, IOException {
+    HdfsState state = injector.getInstance(HdfsState.class);
+    FrameworkID expectedId = createFrameworkId();
+    state.setFrameworkId(expectedId);
+    FrameworkID actualId = state.getFrameworkId();
+    assertEquals(expectedId, actualId);
+  }
+
+  @Test
+  public void testRecordTask()
```

Comment:
I don't see value to this test at all... it uses mostly mocks and test code.  It gives the illusion of value.
The HdfsState is backed by in-memory code that isn't used in prod.  
The first assertion is to make sure the in-memory thing that is never used in prod initializes to 0 tasks.
We create a dumby task, we store it in-memory and pull it out.
Then we assert the tasks are equal...   At best we are testing the serializer which could be done in more clear ways IMO.